default/AI/ResourcesAI.py

```diff
@@ -117,6 +124,87 @@ def print_resources_priority():
+    adjusted_p_defenses = local_p_defenses * (1.0 if currentFocus[pid] != PFocus else 0.5)
+    local_fleet_rating = sys_status.get('myFleetRating', 0)
+    combined_local_defenses = sys_status.get('all_local_defenses', 0)
+    my_neighbor_rating = sys_status.get('my_neighbor_rating', 0)
+    neighbor_threat = sys_status.get('neighborThreat', 0)
+    safety_factor = 1.2 if currentFocus[pid] == PFocus else 0.5
+    cur_shield = this_planet.currentMeterValue(fo.meterType.shield)
+    max_shield = this_planet.currentMeterValue(fo.meterType.maxShield)
+    cur_troops = this_planet.currentMeterValue(fo.meterType.troops)
+    max_troops = this_planet.currentMeterValue(fo.meterType.maxTroops)
+    max_shield = this_planet.currentMeterValue(fo.meterType.maxShield)
+    cur_defense = this_planet.currentMeterValue(fo.meterType.defense)
+    max_defense = this_planet.currentMeterValue(fo.meterType.maxDefense)
+    def_meter_pairs = [(cur_troops, max_troops), (cur_shield, max_shield), (cur_defense, max_defense)]
+    use_protection = True
+    if (fleet_threat and  # i.e., an enemy is sitting on us
+              (currentFocus[pid] != PFocus or  # too late to start protection TODO: but maybe regen worth it
+              # protection forcus only useful here if it maintains an elevated level
+              all([AIDependencies.PROT_FOCUS_MULTIPLIER * a < b for a,b in def_meter_pairs]))):
+        use_protection = False
+    elif ((cur_shield < max_shield - 2 and not tech_is_complete(AIDependencies.PLANET_BARRIER_I_TECH)) and
+            (cur_defense < max_defense - 2 and not tech_is_complete(AIDependencies.DEFENSE_REGEN_1_TECH)) and
+            (cur_troops < max_troops - 2)):
+        use_protection = False
+    elif max(max_shield, max_troops, max_defense) < 3:  # joke defenses, don't bother with protection focus
+        use_protection = False
+    elif regional_threat and local_production_diff <= 2.0:
+        pass  # i.e., use_protection = True
+    elif safety_factor * regional_threat <= local_fleet_rating:
+        use_protection = False
+    elif (safety_factor * regional_threat <= combined_local_defenses and
+              (currentFocus[pid] != PFocus or
+              (0.5 * safety_factor * regional_threat <= local_fleet_rating and
+                   fleet_threat == 0 and neighbor_threat < combined_local_defenses and
+                   local_production_diff > 5))):
+        use_protection = False
+    elif (regional_threat <= FleetUtilsAI.combine_ratings(local_fleet_rating, adjusted_p_defenses) and
+              safety_factor * regional_threat <=
+              FleetUtilsAI.combine_ratings_list([my_neighbor_rating, local_fleet_rating, adjusted_p_defenses])  and
+              local_production_diff > 5):
+        use_protection = False
+    if use_protection or currentFocus[pid] == PFocus:
+        advisory_str = "Advising %sProtection Focus" % ["dropping ", ""][use_protection]
```

Comment:
Here is the lines that is done and what looks better for me.
```python
a = b % c; a += d; return a % e
return (b + d) % (c, e)
```
My suggestion is added as example using `print a,`. It works for me.